lib/adapters/test-result/testplane.ts

```diff
@@ -52,25 +55,161 @@ const wrapSkipComment = (skipComment: string | null | undefined): string => {
     return skipComment ?? 'Unknown reason';
 };
 
-const getHistory = (history?: TestplaneTestResult['history']): TestStepCompressed[] => {
-    return history?.map(h => {
-        const result: TestStepCompressed = {
-            [TestStepKey.Name]: h[TestStepKey.Name],
-            [TestStepKey.Args]: h[TestStepKey.Args],
-            [TestStepKey.Duration]: h[TestStepKey.Duration],
-            [TestStepKey.TimeStart]: h[TestStepKey.TimeStart],
-            [TestStepKey.IsFailed]: h[TestStepKey.IsFailed],
-            [TestStepKey.IsGroup]: h[TestStepKey.IsGroup]
-        };
-
-        if (h[TestStepKey.Children] && h[TestStepKey.IsFailed]) {
-            result[TestStepKey.Children] = getHistory(h[TestStepKey.Children]);
+const testItemsTheSame = (a: TestStepCompressed, b: TestStepCompressed): boolean => (
+    a &&
+    (!a[TestStepKey.Children] || !b[TestStepKey.Children]) &&
+    (!a[TestStepKey.IsFailed] || !b[TestStepKey.IsFailed]) &&
+    a[TestStepKey.Name] === b[TestStepKey.Name] &&
+    a[TestStepKey.Args].join() === b[TestStepKey.Args].join()
+);
+
+const arraysEqual = <I>(a: I[], b: I[], checkFunc: (a: I, b: I) => boolean): boolean => {
```

Comment:
I personally don't like such very generic and "universal" helpers, imho they don't bring much value beyond core APIs, and are often forgotten about the minute this PR is merged. FWIW, we already have lodash as a dependency and if you don't want to compare length twice so much, you could use `_.isEqualWith`

But that's just my personal opinion, you don't have to fix this. It's a matter of taste.

Reply:
It is better than is _. isEqualWith because check also length